pkg/data/course.go

```diff
@@ -0,0 +1,77 @@
+package data
+
+import (
+	"fmt"
+
+	"gorm.io/gorm"
+)
+
+type Course struct {
+	Code           int    `gorm:"code"`
+	Title          string `gorm:"title"`
+	DepartmentCode int    `gorm:"department_code"`
+	Description    string `gorm:"description"`
+}
+
+type CourseData struct {
+	db *gorm.DB
+}
+
+func NewCourseData(db *gorm.DB) *CourseData {
+	return &CourseData{db: db}
+}
+
+func (u CourseData) Add(course Course) (int, error) {
```

Comment:
You have struct the `CourseData`, why you decided to name this variable `u`?
Please rewrite it to `cd` or just `c`